tests/test_integration.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path
+import re
+
+import pytest
+
+import build.__main__
+
+
+_SDIST = re.compile('.*.tar.gz')
+_WHEEL = re.compile('.*.whl')
+
+
+@pytest.mark.parametrize(
+    ('project'),
+    [
+        'python-build',
+        'pip',
+        'dateutil',
+        'Solaar',
+    ]
+)
+@pytest.mark.parametrize(
+    ('args'),
+    [
+        [],
+        ['-x', '--no-isolation'],
+    ]
+)
+def test_build(tmpdir, monkeypatch, integration_path, project, args):
+    monkeypatch.setenv('SETUPTOOLS_SCM_PRETEND_VERSION', 'dummy')  # for the projects that use setuptools_scm
+
+    path = os.path.join(integration_path, project)
+
+    build.__main__.main([path, '-o', tmpdir] + args)
+
+    assert filter(_SDIST.match, os.listdir(tmpdir))
+    assert filter(_WHEEL.match, os.listdir(tmpdir))
+
+
+def test_isolation(tmpdir, test_flit_path, mocker):
+    try:
+        import flit_core  # noqa: F401
```

Comment:
Why would flit core not be always available?

Reply:
It's the opposite, if flit is available we can't properly test the isolation, so we skip it. The test won't run and show as missing coverage.

Reply:
I put the xfail in the `try` instead of the `else` and added a comment. This should clarify things.